### PR TITLE
Added MasterKeyLoader abstraction

### PR DESCRIPTION
**Pull request by Zhaars**
One of the key concepts with AcraMasterKeyLoaders(AMK) mentioned by @Lagovas is to have enough flexibility with AcraEE.

Possible scenarios could be when we want to extend some functionality related to AMKL in AcraEE(added new KMS, or have some custom cases). So this PR contains some draft ideas to have some kind of abstraction that helps us have the generic approach to creating MKL.

The key thing is to have `MasterKeyLoaderFactory` interface that can create any `MasterKeyLoader`.

```
type MasterKeyLoaderFactory interface {
	CreateMasterKeyLoader() (MasterKeyLoader, error)
}
```

`MasterKeyLoaderCreator` - represent the current implementation of  `MasterKeyLoaderFactory`  which depends on provided `loadStrategy`. So if need to have some custom behavior in AcraEE we need to reimplement the EE version of `MasterKeyLoaderCreator` and the remaining code should be as is.

P.S Feel free to suggest any better naming or ideas.

## Checklist

- [ ] Change is covered by automated tests
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes
- [ ] CHANGELOG.md is updated (in case of notable or breaking changes)
- [ ] CHANGELOG_DEV.md is updated
- [ ] Benchmark results are attached (if applicable)
- [ ] [Example projects and code samples] are up-to-date (in case of API changes) 

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: cmd/acra-keys/keys/generate.go [hunk 1]
```diff
@@ -298,7 +298,8 @@ func (g *GenerateKeySubcommand) Execute() {
 		}
 	}
 
-	keyLoader, err := keyloader.GetInitializedMasterKeyLoader(g.CommonKeyStoreParameters.KeyLoaderCLIOptions().KeystoreEncryptorType)
+	masterKeyLoaderFactory := keyloader.NewMasterKeyLoaderFactory(g.CommonKeyStoreParameters.KeyLoaderCLIOptions().KeystoreEncryptorType)
+	keyLoader, err := keyloader.GetInitializedMasterKeyLoader(masterKeyLoaderFactory)
```
Comment (G1gg1L3s) [inline cmd/acra-keys/keys/generate.go:302]: This pattern repeats a couple of times. Are there any reasons not to move it into a separate function? Like `keyloader.{Build, Create, New, Assemble}MasterKeyLoader(loadStrategy string)`?

**Comment by Lagovas** [comment]
There is open question, how we can replace existing implementation with new one?
For example in acra-ee we will create new implementation in file `acra/keystore/keyloader/ee_loader.go`:

```
type EEMasterKeyLoaderCreator struct {
	loadStrategy string
	envName      string
}

func NewEEMasterKeyLoaderFactory(loadStrategy string) MasterKeyLoaderFactory {
	return EEMasterKeyLoaderCreator{
		loadStrategy: loadStrategy,
		envName:      keystore.AcraMasterKeyVarName,
	}
}
```

How it will replace default implementation in `acra-server.go` ?

**Comment by Zhaars** [comment]
> There is open question, how we can replace existing implementation with new one? For example in acra-ee we will create new implementation in file `acra/keystore/keyloader/ee_loader.go`:
> 
> ```
> type EEMasterKeyLoaderCreator struct {
> 	loadStrategy string
> 	envName      string
> }
> 
> func NewEEMasterKeyLoaderFactory(loadStrategy string) MasterKeyLoaderFactory {
> 	return EEMasterKeyLoaderCreator{
> 		loadStrategy: loadStrategy,
> 		envName:      keystore.AcraMasterKeyVarName,
> 	}
> }
> ```
> 
> How it will replace default implementation in `acra-server.go` ?

@Lagovas Can we do a similar trick as we did with `aws_enable` via build flags? I mean switch on standard implementation by default but when we want to use another ee version switch off standard and switch on ee version/

**Review by Lagovas — COMMENTED** [review]
lgtm in general. lets fix tests and merge

**Review by Lagovas — APPROVED** [review]

**Pull request merged**